src/core/lib/CipherSaber2.mjs

```diff
@@ -4,7 +4,7 @@
  * @license Apache-2.0
  */
 export function encode(tempIVP, key, rounds, input) {
-    const ivp = new Uint8Array(key.concat(tempIVP));
+    const ivp = new Uint8Array([...key, ...tempIVP]);
```

Comment:
key is a `number[]` and tempIVP is a `Buffer`, so I was initially surprised that they can't be concatenated, but a quick test shows it's true:

```javascript
> const tempIVP = Buffer.from([1, 2])
undefined
> const key = [3, 4]
undefined
> key.concat(tempIVP)
[ 3, 4, <Buffer 01 02> ]
```

An alternative seems to be spreading _just_ the buffer (e.g. `key.concat(...tempIVP)`), but explicitly spreading both seems like a nicer solution anyway.  Nice catch!